cmd/translate/main_grpc_integration_test.go

```diff
@@ -672,6 +674,272 @@ func Test_ListTranslations_gRPC(t *testing.T) {
+					},
+					{
+						Id: "Welcome", Message: "Hi", Description: "greeting", Status: translatev1.Message_UNTRANSLATED,
+					},
+				},
+			},
+				&field_mask.FieldMask{Paths: []string{"messages"}}),
+			translation: createTranslation(ctx, t, service6.GetId(), &translatev1.Translation{
+				Original: false,
+				Language: langs[0].String(),
+				Messages: []*translatev1.Message{
+					{
+						Id: "Hello", Message: "Planet",
+					},
+				},
+			}),
+		},
+		{
+			name: "Add multiple messages to original translation",
+			request: randUpdateTranslationReq(t, service7.GetId(), &translatev1.Translation{
+				Language: langs[0].String(),
+				Messages: []*translatev1.Message{
+					{
+						Id: "Bye", Message: "World",
+					},
+					{
+						Id: "Animal", Message: "Dog",
+					},
+					{
+						Id: "Welcome", Message: "Hi",
+					},
+				},
+			},
+				&field_mask.FieldMask{Paths: []string{"messages"}}),
+			translation: createTranslation(ctx, t, service7.GetId(), &translatev1.Translation{
+				Original: true,
+				Language: langs[0].String(),
+				Messages: []*translatev1.Message{
+					{
+						Id: "Hello", Message: "Planet",
+					},
+				},
+			}),
+		},
+		{
+			name: "Add messages, update description, message and status",
+			request: randUpdateTranslationReq(t, service8.GetId(), &translatev1.Translation{
+				Language: langs[0].String(),
+				Messages: []*translatev1.Message{
+					{
+						Id: "Bye", Message: "World", Description: "farewell", Status: translatev1.Message_UNTRANSLATED,
+					},
+					{
+						Id: "Hi", Message: "Dog", Description: "greeting", Status: translatev1.Message_UNTRANSLATED,
+					},
+					{
+						Id: "Hello", Message: "Hi", Description: "greeting", Status: translatev1.Message_UNTRANSLATED,
+					},
+				},
+			},
+				&field_mask.FieldMask{Paths: []string{"messages"}}),
+			translation: createTranslation(ctx, t, service8.GetId(), &translatev1.Translation{
+				Original: false,
+				Language: langs[0].String(),
+				Messages: []*translatev1.Message{
+					{
+						Id: "Hello", Message: "Planet", Description: "A simple greeting", Status: translatev1.Message_TRANSLATED,
+					},
+				},
+			}),
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		subtest(tt.name, func(ctx context.Context, t *testing.T) {
+			beforeUpdateTranslations, err := client.ListTranslations(ctx, &translatev1.ListTranslationsRequest{
+				ServiceId: tt.request.GetServiceId(),
+			})
+
+			require.NoError(t, err)
+			resp, err := client.UpdateTranslation(ctx, tt.request)
+
+			if err == nil {
+				require.NotNil(t, resp)
+			}
+
+			afterUpdateTranslations, err := client.ListTranslations(ctx, &translatev1.ListTranslationsRequest{
+				ServiceId: tt.request.GetServiceId(),
+			})
+
+			require.NoError(t, err)
+			require.NotEmpty(t, resp)
+
+			var translationFromService *translatev1.Translation
+
+			for i := range afterUpdateTranslations.GetTranslations() {
```

Comment:
Never let actual values guide your tests. Use your expectations to guide.

E.g. `ListTranslations` returns an empty list, the test passes, right? But, actually, it should fail.